experimental/dds/tree2/src/core/repair/repairDataBuilder.ts

```diff
@@ -0,0 +1,46 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { fail } from "../../util";
+import { ChangeAtomId } from "../rebase";
+import { FieldKey } from "../schema-stored";
+import { Delta } from "../tree";
+
+/**
+ * A function that associates a change that produces repair data with the detached
+ * field where the repair data should be stored.
+ * @alpha
+ */
+export type RepairDataHandler = (changeId: ChangeAtomId) => FieldKey;
+
+/**
+ * TODO doc
+ * @alpha
+ */
+export interface RepairDataBuilder {
+	handler: RepairDataHandler;
+	/**
+	 * A map of detached fields to the delta that would move in the appropriate repair data.
+	 */
+	marks: Map<FieldKey, Delta.MarkList>;
```

Comment:
Exposing the map to the code that adds content into it means they could theoretically remove content from it or condition their behavior based on what else is in the map. Consider tightening the contract with something like: `type RepairDataAccumulator = (key: FieldKey, delta: Delta.MarkList) => void`. You can still use a map behind the scenes.